### ISSUE TITLE
RF-DETR Nano huge differnce in training quality when trained locally vs trained on roboflow.com

### ISSUE THREAD

**Issue by MichalTurek** [roboflow/rf-detr#484]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

I am working on my Master's thesis related to object detection.  One goal is to compare how different architectures perform when trained on limited hardware (8GB VRAM). The results on roboflow.com using DETR-NANO look impressive:

<img width="1663" height="283" alt="Image" src="https://github.com/user-attachments/assets/0c46544d-b9ff-4d00-8134-b87bd036b8d2" />

However when running training script:

I notice that when trained on roboflow.com the model immidietly starts from ~0.5 MAP and achives far better result: 

<img width="1980" height="682" alt="Image" src="https://github.com/user-attachments/assets/c9e55e2c-cb44-4cc7-8c72-78046ac54cbf" />
 When trained locally: 

<img width="1483" height="737" alt="Image" src="https://github.com/user-attachments/assets/e079eebd-13ac-48cf-8997-c8d2eef63311" />

<img width="1488" height="695" alt="Image" src="https://github.com/user-attachments/assets/c84772cf-5e57-431c-a7a1-1379f8eee0d6" />

## Python
Version: 3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)]
Executable: C:\Program Files\Python310\python.exe
Implementation: CPython
Prefix: C:\Program Files\Python310

System: Windows
Release: 10
Version: 10.0.26200
Machine: AMD64
Processor: Intel64 Family 6 Model 141 Stepping 1, GenuineIntel
Platform string: Windows-10-10.0.26200-SP0

## RF-DETR
rfdetr module: <module 'rfdetr' from 'C:\\Users\\micha\\AppData\\Roaming\\Python\\Python310\\site-packages\\rfdetr\\__init__.py'>
rfdetr (importlib.metadata): 1.3.0
rfdetr module file: C:\Users\micha\AppData\Roaming\Python\Python310\site-packages\rfdetr\__init__.py

## PyTorch / CUDA
torch.__version__: 2.9.0+cu130
torch.version.cuda: 13.0
torch.backends.cudnn.version(): 91200
CUDA available (torch.cuda.is_available()): True
Device count: 1
- Device 0: NVIDIA GeForce RTX 3070 Laptop GPU, capability (8, 6)

## Key Python packages
rfdetr==1.3.0
torch==2.9.0+cu130
torchvision==0.24.0+cu130
torchaudio: not installed or not found
transformers==4.57.3
timm==1.0.22

## pip freeze

absl-py==2.3.1
accelerate==1.12.0
annotated-types==0.7.0
anywidget==0.9.21
asttokens==3.0.1
bbox_visualizer==0.2.2
certifi==2025.10.5
charset-normalizer==3.4.4
colorama==0.4.6
comm==0.2.3
contourpy==1.3.2
cycler==0.12.1
Cython==3.2.2
decorator==5.2.1
defusedxml==0.7.1
einops==0.8.1
exceptiongroup==1.3.1
executing==2.2.1
fairscale==0.4.13
filelock==3.19.1
filetype==1.2.0
filterpy==1.4.5
fonttools==4.60.1
fsspec==2025.9.0
ftfy==6.3.1
gitdb==4.0.12
GitPython==3.1.45
grpcio==1.76.0
huggingface-hub==0.36.0
idna==3.7
ipython==8.37.0
ipywidgets==8.1.8
jedi==0.19.2
Jinja2==3.1.6
joblib==1.5.2
jupyter_bbox_widget==0.6.0
jupyterlab_widgets==3.0.16
kiwisolver==1.4.9
lap==0.5.12
Markdown==3.10
markdown-it-py==4.0.0
MarkupSafe==2.1.5
matplotlib==3.10.7
matplotlib-inline==0.2.1
mdurl==0.1.2
mpmath==1.3.0
networkx==3.3
ninja==1.13.0
norfair==2.3.0
numpy==2.2.6
open_clip_torch==3.2.0
opencv-python==4.12.0.88
opencv-python-headless==4.10.0.84
packaging==25.0
pandas==2.3.3
parso==0.8.5
peft==0.18.0
pi_heif==1.1.1
pillow==11.3.0
pillow-avif-plugin==1.5.2
polars==1.35.1
polars-runtime-32==1.35.1
polygraphy==0.49.26
prompt_toolkit==3.0.52
protobuf==6.33.1
psutil==7.1.3
psygnal==0.15.0
pure_eval==0.2.3
pycocotools==2.0.10
pydantic==2.12.5
pydantic_core==2.41.5
Pygments==2.19.2
pylabel==0.1.55
pyparsing==3.2.5
python-dateutil==2.9.0.post0
python-dotenv==1.2.1
pytz==2025.2
PyYAML==6.0.3
regex==2025.11.3
requests==2.32.5
requests-toolbelt==1.0.0
rf100vl==1.1.0
rfdetr==1.3.0
rich==14.2.0
roboflow==1.2.11
safetensors==0.7.0
scikit-learn==1.7.2
scipy==1.15.3
seaborn==0.13.2
six==1.17.0
smmap==5.0.2
stack-data==0.6.3
supervision==0.27.0
sympy==1.14.0
tensorboard==2.20.0
tensorboard-data-server==0.7.2
threadpoolctl==3.6.0
timm==1.0.22
tokenizers==0.22.1
torch==2.9.0+cu130
torchvision==0.24.0+cu130
tqdm==4.67.1
traitlets==5.14.3
transformers==4.57.3
typing-inspection==0.4.2
typing_extensions==4.15.0
tzdata==2025.2
ultralytics==8.3.223
ultralytics-thop==2.0.18
urllib3==2.5.0
wcwidth==0.2.14
Werkzeug==3.1.4
widgetsnbextension==4.0.15

### Minimal Reproducible Example
```

`
def main():
    model = RFDETRNano()
    model.maybe_download_pretrain_weights()
    model.train(
        dataset_dir=DATASET_DIR,
        output_dir=OUTPUT_DIR,
        epochs=300,
        batch_size=16,
        grad_accum_steps=1,
        resolution=384,
        device="cuda",

        dataloader_num_workers=4,   
        # --- EARLY STOPPING ---
        early_stopping=True,
        early_stopping_patience=20,      
        early_stopping_min_delta=0.0,   
)


if __name__ == "__main__":
    import torch.multiprocessing as mp
    mp.freeze_support()  
    main()
` 
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by isaacrob-roboflow** [comment]
Weird. The major difference is we use the o365 checkpoint on platform but this repo has coco. O365 should give slightly better results. What you're showing looks like a serious learning instability. How does your data look? Also, how did you generate those plots? Ideally you should be using the EMA weights

**Comment by MichalTurek** [comment]
The data is a rader difficult one, in this particular training its a ball detection on football field. I know it is a small object and it maybe the reason but to be fair if training on roboflow site works properly why would it not work here ?

**Comment by MichalTurek** [comment]
> Weird. The major difference is we use the o365 checkpoint on platform but this repo has coco. O365 should give slightly better results. What you're showing looks like a serious learning instability. How does your data look? Also, how did you generate those plots? Ideally you should be using the EMA weights
if that is the only difference, is there a way to use other checkpoint to test it out?

**Comment by MichalTurek** [comment]
<img width="1475" height="1182" alt="Image" src="https://github.com/user-attachments/assets/9a39ecd6-d0b6-4d58-b668-ccf48d537481" /> it is essentialy the same story with rerun...

**Comment by isaacrob-roboflow** [comment]
Wild. I don't have a great explanation, hard to make any guesses without seeing data. 

Do you mind running more experiments? I'd try bs 2 and grad accum 8, and separately lr 1e-5

It is interesting that the peak score looks similar

**Comment by isaacrob-roboflow** [comment]
Oh please use the latest version of the repo from source. There are some improvements. 

When training on platform, what resize did you set in your dataset version?

**Comment by MichalTurek** [comment]
Yeah, that is what I am going to do anyway. Saw in your paper discussion about the problem of how others report their results. But do you have any suspicion as to why the local result is just way off the remote one? 
The message which I sent was only about the fact that I want to do fair comparison and by looking at the remote results I know that your model is indeed better, but when I did side by side comparison using some match video I see that currently YoloV11 tends to lead (compared to currently trained DETR model, which I assume has some issue)

**Comment by MichalTurek** [comment]
> Oh please use the latest version of the repo from source. There are some improvements.
> 
> When training on platform, what resize did you set in your dataset version?

I didnt do any resize of dataset, so it is by default all images between 720p & 1080p but It said that the model is sized to be 384x384 anyway.
Got is, so the next step would be to rerun on latest version of code on develop branch?

**Comment by isaacrob-roboflow** [comment]
Yeah use latest from dev. If you don't specify a resize on roboflow platform it will do 640 by default which is not the appropriate resolution for RF-DETR. A quirk of the platform that we're trying to figure out how to address

**Comment by MichalTurek** [comment]
Oh that is interesting, so it may be better due to the fact that it uses higher resolution (640x640 no 384x384) which for my case (detecting tiny ball) is highly beneficial?

**Comment by isaacrob-roboflow** [comment]
The model's resolution gets set to 640

**Comment by MichalTurek** [comment]
Okay then it tottally make sense, It is not a bug on the roboflow repo, its more about the misinformation on roboflow platform, as it mentioned that the train resolution for nano will be 384x384. For this particular dataset, with really tiny objects to be detected higer resolution improves quality largely
